sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -714,6 +729,57 @@ public void close() throws IOException {
     }
   }
 
+  /**
+   * This is the default predicate used to test if a failed ES operation should be retried. A retry
+   * will be attempted until the maxAttempts or maxDuration is exceeded, whichever comes first, for
+   * TOO_MANY_REQUESTS(429) exception.
+   */
+  static class DefaultRetryPredicate implements RetryPredicate {
```

Comment:
This is package private, but you have modified the configuration to require users to provide this. I expect that won't work outside of the package will it?

See previous comment where I recommend not letting users provide this at all though.